Sources/SmileID/Classes/Networking/SmileIDService.swift

```diff
@@ -97,35 +97,33 @@ public extension SmileIDServiceable {
     ///   - numAttempts: The maximum number of polls before ending the flow
     func pollJobStatus<T: JobResult>(
         request: JobStatusRequest,
-        interval _: TimeInterval,
+        interval: TimeInterval,
         numAttempts: Int
-    ) async throws -> JobStatusResponse<T> {
-        var lastError: Error?
-        var attemptCount = 0
-
-        func makeRequest() async throws -> JobStatusResponse<T> {
-            attemptCount += 1
-
-            do {
-                let response: JobStatusResponse<T> = try await SmileID.api.getJobStatus(request: request)
-                if response.jobComplete {
-                    return response
-                } else if attemptCount < numAttempts {
-                    return try await makeRequest()
-                } else {
-                    throw SmileIDError.jobStatusTimeOut
+    ) -> AsyncThrowingStream<JobStatusResponse<T>, Error> {
+        AsyncThrowingStream { continuation in
+            Task {
+                var latestError: Error?
+                for _ in 0..<numAttempts {
+                    do {
+                        let response: JobStatusResponse<T> = try await SmileID.api.getJobStatus(request: request)
+                        continuation.yield(response)
+                        // Reset the error if the API response was successful
+                        latestError = nil
+                        if response.jobComplete {
+                            break
+                        }
+                    } catch {
+                        latestError = error
+                    }
+                    try await Task.sleep(nanoseconds: UInt64(interval * 1_000_000_000))
```

Comment:
Could you please explain why you're introducing this delay? I think this can slow down the code execution.

Reply:
Already discussed this during our sync.